tests/integration/targets/rabbitmq_feature_flag/tasks/tests.yml

```
@@ -0,0 +1,19 @@
+- block:
+  - set_fact:
+      parameter_name: maintenance_mode_status
+      parameter_node: rabbit
+
+  - name: Enable feature flag
+    rabbitmq_feature_flag:
+      name: "{{ parameter_name }}"
+      node: "{{ parameter_node }}"
+    register: result
+
+  - name: Check if feature_flag module was completed successfully
+    assert:
+      that:
+        - result is success
```

```suggestion
        - result is success
        - result is changed
```
Please:
1. Run this first in check mode
2. Check the module reports changed
3. Check that nothing has actually changed
4. Run this in real mode
5. Check the module reports changed
6. Check that the state has actually changed
7. Run this again in check mode
8. Check that the module reports that nothing has changed
9. Run this again in real mode
10. The module report nothing has changed

---

unfortunately i have no idea how to run tests in check mode :/

And I can't just add 'result is changed' to the assert, because in latest rabbitmq versions, this flag is enabled by default

---

add similar tasks with `check_mode: yes` and check that they'll report correct things.

---

I'm not sure if i got Your idea correctly but I've covered check_mode in integration tests

---

yes, looks like now it's covered as well, thanks!